src/clj_okhttp/formats.clj

```diff
@@ -35,29 +37,40 @@
                                (okhttp/->request-body content-type (encoder body (.-charset negotiated)))
                                (throw (ex-info (str "No muuntaja encoder defined for " content-type) {}))))))]
     (cond-> request
-      (some? request-body) (assoc :body request-body)
-      (some? content-type) (assoc-in [:headers "content-type"] content-type))))
-
-(defn resolve-content-type [content-type]
-  (case content-type
-    :stream "application/octet-stream"
-    :csv "text/csv"
-    :json "application/json"
-    :yaml "application/yaml"
-    :edn "application/edn"
-    :clojure "application/edn"
-    :transit "application/transit+json"
-    (or content-type "application/octet-stream")))
-
-(defn format-stream [muuntaja content-type stream]
-  (let [resolved   (resolve-content-type content-type)
-        negotiated ^FormatAndCharset (negotiate-content-type muuntaja resolved)]
-    (if-some [decoder (m/decoder muuntaja (.-format negotiated))]
-      (decoder stream (.-charset negotiated))
-      stream)))
+            (some? request-body) (assoc :body request-body)
+            (some? content-type) (assoc-in [:headers "content-type"] content-type))))
+
+(def coercion-mapping
+  {:csv     "text/csv"
+   :json    "application/json"
+   :yaml    "application/yaml"
+   :edn     "application/edn"
+   :clojure "application/edn"
+   :transit "application/transit+json"})
+
+(defn coerce-stream [^InputStream stream coercion detected-charset]
+  (case coercion
+    :text (slurp stream :encoding detected-charset)
```

Comment:
I think if we add support to muuntaja for "text/plain" that we can treat this like any other coercion-mapping.

Reply:
See my comment https://github.com/RutledgePaulV/clj-okhttp/issues/3#issuecomment-1030745224 to continue the conversation there.